### PR TITLE
Add cluster channel modal

### PR DESCRIPTION
**Pull request by rhamilto**
https://jira.coreos.com/browse/CONSOLE-1062

vjupfp0kzr

### CODE REVIEW

File: frontend/public/components/cluster-settings/cluster-settings.tsx [hunk 1]
```diff
@@ -31,6 +31,12 @@ const MOCK_CLUSTER_UPDATE = localStorage.getItem('MOCK_CLUSTER_UPDATE');
 const clusterAutoscalerReference = referenceForModel(ClusterAutoscalerModel);
 const clusterVersionReference = referenceForModel(ClusterVersionModel);
 
+export const getAvailableClusterChannels = () => {return {fast: 'fast', nightly: 'nightly', production: 'production'}};
```
Comment (spadgett) [inline frontend/public/components/cluster-settings/cluster-settings.tsx]:
```suggestion
export const getAvailableClusterChannels = () => ({fast: 'fast', nightly: 'nightly', production: 'production'});
```

File: frontend/public/components/cluster-settings/cluster-settings.tsx [hunk 2]
```diff
@@ -136,6 +141,12 @@ const UpdateStatus: React.SFC<UpdateStatusProps> = ({cv}) => {
   </React.Fragment>;
 };
 
+// TODO:  origin has no channels, so channel selection will not apply
+// How to determine if origin?
+const CurrentChannel: React.SFC<CurrentChannelProps> = ({cv}) => <button className="btn btn-link co-m-modal-link" onClick={() => (clusterChannelModal({cv}))}>
+  {getCurrentClusterChannel(cv)}
```
Comment (spadgett) [inline frontend/public/components/cluster-settings/cluster-settings.tsx]: I'd probably just use

```suggestion
  {cv.spec.channel || '-'}
```

File: frontend/public/components/modals/cluster-channel-modal.tsx [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+  constructor(public props: ClusterChannelModalProps) {
+    super(props);
+  }
+
+  _submit = (e) => {
+    e.preventDefault();
+    const {cv} = this.props;
+    const {selectedChannel} = this.state;
+    const desiredChannel = getAvailableClusterChannels()[selectedChannel];
```
Comment (spadgett) [inline frontend/public/components/modals/cluster-channel-modal.tsx]: Is this different from `selectedChannel`?

File: frontend/public/components/modals/cluster-channel-modal.tsx [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+    const availableChannels = getAvailableClusterChannels();
+    const currentChannel = getCurrentClusterChannel(cv);
+    return <form onSubmit={this._submit} name="form">
+      <ModalTitle>Update Channel</ModalTitle>
+      <ModalBody>
+        <div className="form-group">
+          <label htmlFor="channel_dropdown">Channel</label>
+          <Dropdown
+            className="cluster-channel-modal__dropdown"
+            id="channel_dropdwon"
```
Comment (spadgett) [inline frontend/public/components/modals/cluster-channel-modal.tsx]: typo

```suggestion
            id="channel_dropdown"
```

**Comment by spadgett** [comment]
Looks good. We just need the exception to be approved to remove the hold.

/assign @TheRealJon @benjaminapetersen 

Jon, can you take over this PR since Robb is out? It should just be getting it through CI and removing the hold when we can. @benjaminapetersen is on the exception mailing list thread.

**Comment by TheRealJon** [comment]
@spadgett Will do

**Comment by benjaminapetersen** [comment]
/hold remove

We have approval on this.

**Comment by benjaminapetersen** [comment]
i forget what the comment is, gimme a min 😄

**Comment by benjaminapetersen** [comment]
ha, got it.

**Comment by benjaminapetersen** [comment]
👍

**Pull request merged**